core/dbt/config/renderer.py

```diff
@@ -109,10 +109,14 @@ def __init__(
         # even when we don't have a profile.
         if cli_vars is None:
             cli_vars = {}
+        # Store profile and cli_vars for creating strict context later
+        self.profile = profile
+        self.cli_vars = cli_vars
+        # Lenient context for project loading (dbt deps, dbt_project.yml)
         if profile:
-            self.ctx_obj = TargetContext(profile.to_target_dict(), cli_vars)
+            self.ctx_obj = TargetContext(profile.to_target_dict(), cli_vars, require_vars=False)
         else:
-            self.ctx_obj = BaseContext(cli_vars)  # type:ignore
+            self.ctx_obj = BaseContext(cli_vars, require_vars=False)  # type:ignore
```

Comment:
This here means we're _never_ requiring vars during the rendering of the `dbt_project.yaml`. Is that what we want? Reading the PR description it sounds, and the original issue, I think we only want to not require vars during the rendering of the `dbt_project.yml` when running `deps`, `clean`, or `debug`. Please correct me if I'm misinterpreting things 😅